ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/view/ToolAdminViewImpl.java

```diff
@@ -0,0 +1,242 @@
+package org.iplantc.de.admin.desktop.client.toolAdmin.view;
+
+import org.iplantc.de.admin.desktop.client.toolAdmin.ToolAdminView;
+import org.iplantc.de.admin.desktop.client.toolAdmin.events.AddToolSelectedEvent;
+import org.iplantc.de.admin.desktop.client.toolAdmin.events.DeleteToolSelectedEvent;
+import org.iplantc.de.admin.desktop.client.toolAdmin.events.SaveToolSelectedEvent;
+import org.iplantc.de.admin.desktop.client.toolAdmin.events.ToolSelectedEvent;
+import org.iplantc.de.admin.desktop.client.toolAdmin.model.ToolProperties;
+import org.iplantc.de.admin.desktop.client.toolAdmin.view.cells.ToolAdminNameCell;
+import org.iplantc.de.admin.desktop.client.toolAdmin.view.dialogs.ToolAdminDetailsDialog;
+import org.iplantc.de.client.models.tool.Tool;
+import org.iplantc.de.commons.client.ErrorHandler;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.event.logical.shared.ValueChangeEvent;
+import com.google.gwt.event.shared.HandlerRegistration;
+import com.google.gwt.inject.client.AsyncProvider;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiFactory;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.uibinder.client.UiHandler;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+import com.google.inject.assistedinject.Assisted;
+
+import com.sencha.gxt.core.client.IdentityValueProvider;
+import com.sencha.gxt.core.client.Style;
+import com.sencha.gxt.data.shared.ListStore;
+import com.sencha.gxt.data.shared.Store;
+import com.sencha.gxt.widget.core.client.Composite;
+import com.sencha.gxt.widget.core.client.Dialog;
+import com.sencha.gxt.widget.core.client.box.ConfirmMessageBox;
+import com.sencha.gxt.widget.core.client.button.TextButton;
+import com.sencha.gxt.widget.core.client.event.DialogHideEvent;
+import com.sencha.gxt.widget.core.client.event.SelectEvent;
+import com.sencha.gxt.widget.core.client.grid.ColumnConfig;
+import com.sencha.gxt.widget.core.client.grid.ColumnModel;
+import com.sencha.gxt.widget.core.client.grid.Grid;
+
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * @author aramsey
+ */
+public class ToolAdminViewImpl extends Composite implements ToolAdminView {
+
+    interface ToolAdminViewImplUiBinder extends UiBinder<Widget, ToolAdminViewImpl> {
+
+    }
+
+    private final class NameFilter implements Store.StoreFilter<Tool> {
+
+        private String filterText;
+
+        @Override
+        public boolean select(Store<Tool> store, Tool parent, Tool item) {
+            return !Strings.nullToEmpty(filterText).isEmpty() && item.getName()
```

Comment:
FYI: `Strings.nullToEmpty(filterText).isEmpty()` can be reduced to `Strings.isNullOrEmpty(filterText)`.